src/track/trackrecord.cpp

```diff
@@ -102,37 +115,39 @@ void TrackRecord::mergeImportedMetadata(
                 !importedTrackInfo.getTrackNumber().isEmpty()) {
             pMergedTrackInfo->setTrackNumber(importedTrackInfo.getTrackNumber());
+            modified = true;
         }
     }
 #if defined(__EXTRA_METADATA__)
-    copyIfNotNull(mergedTrackInfo.refConductor(), importedTrackInfo.getConductor());
-    copyIfNotNull(mergedTrackInfo.refDiscNumber(), importedTrackInfo.getDiscNumber());
-    copyIfNotNull(mergedTrackInfo.refDiscTotal(), importedTrackInfo.getDiscTotal());
-    copyIfNotNull(mergedTrackInfo.refEncoder(), importedTrackInfo.getEncoder());
-    copyIfNotNull(mergedTrackInfo.refEncoderSettings(), importedTrackInfo.getEncoderSettings());
-    copyIfNotNull(mergedTrackInfo.refISRC(), importedTrackInfo.getISRC());
-    copyIfNotNull(mergedTrackInfo.refLanguage(), importedTrackInfo.getLanguage());
-    copyIfNotNull(mergedTrackInfo.refLyricist(), importedTrackInfo.getLyricist());
-    copyIfNotNull(mergedTrackInfo.refMood(), importedTrackInfo.getMood());
-    copyIfNotNull(mergedTrackInfo.refMovement(), importedTrackInfo.getMovement());
-    copyIfNotNull(mergedTrackInfo.refMusicBrainzArtistId(), importedTrackInfo.getMusicBrainzArtistId());
-    copyIfNotNull(mergedTrackInfo.refMusicBrainzRecordingId(), importedTrackInfo.getMusicBrainzRecordingId());
-    copyIfNotNull(mergedTrackInfo.refMusicBrainzReleaseId(), importedTrackInfo.getMusicBrainzReleaseId());
-    copyIfNotNull(mergedTrackInfo.refMusicBrainzWorkId(), importedTrackInfo.getMusicBrainzWorkId());
-    copyIfNotNull(mergedTrackInfo.refRemixer(), importedTrackInfo.getRemixer());
-    copyIfNotEmpty(mergedTrackInfo.refSeratoTags(), importedTrackInfo.getSeratoTags());
-    copyIfNotNull(mergedTrackInfo.refSubtitle(), importedTrackInfo.getSubtitle());
-    copyIfNotNull(mergedTrackInfo.refWork(), importedTrackInfo.getWork());
-    AlbumInfo& mergedAlbumInfo = refMetadata().refAlbumInfo();
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrConductor(), importedTrackInfo.getConductor());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrDiscNumber(), importedTrackInfo.getDiscNumber());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrDiscTotal(), importedTrackInfo.getDiscTotal());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrEncoder(), importedTrackInfo.getEncoder());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrEncoderSettings(), importedTrackInfo.getEncoderSettings());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrISRC(), importedTrackInfo.getISRC());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrLanguage(), importedTrackInfo.getLanguage());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrLyricist(), importedTrackInfo.getLyricist());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrMood(), importedTrackInfo.getMood());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrMovement(), importedTrackInfo.getMovement());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrMusicBrainzArtistId(), importedTrackInfo.getMusicBrainzArtistId());
```

Comment:
Unfortunately these line are exceeding the GitHub diff view. Please break them, or use the our GitHub pre-comit hook.

Reply:
I tried to keep the formatting to reduce the risk for merge conflicts. Now exactly that happened during reformatting ;) Fixed.